pylint/checkers/base/basic_error_checker.py

```diff
@@ -78,6 +78,36 @@ def _has_abstract_methods(node: nodes.ClassDef) -> bool:
     return len(utils.unimplemented_abstract_methods(node)) > 0
 
 
+def _new_instantiates_super(node: nodes.ClassDef) -> bool:
+    """Check if node implements `__new__`.
+
+    If `__new__` is implemented, check if it calls `super().__new__(cls)`.
+    """
+    if "__new__" not in node.locals:
+        return False
+
+    new = next(node.igetattr("__new__"))
```

Comment:
This should probably be done in a `try...except`. Or with `safe_infer`.

Reply:
```
    try:
        new = next(node.igetattr("__new__"))
    except astroid.InferenceError:
        return False
```

is the closest I see to other repo code using igetattr. However, I can't think of a single example where we'd actually raise `InferenceError` since __new__ is a dunder for pretty much any class. Do you have a suggestion? I can remove the try/except or ignore coverage for this line just for precaution.

Reply:
Hmm.. @Pierre-Sassoulas what do you want? Non-covered code? Or code that is known to raise a certain exception that we just don't know the edge case for yet?